File: src/main/kotlin/entityAndDataSourcesPortfolio/Project.kt
package com.example.entityAndDataSourcesPortfolio

import kotlinx.serialization.Serializable
import org.bson.codecs.pojo.annotations.BsonId
import org.bson.types.ObjectId

@Serializable
data class Project(
    val id: String = java.util.UUID.randomUUID().toString(),
    val title: String,
    val description: String,
    val imageUrl: List<String>,
    val tags: List<String>,
    val projectUrl: String
)

File: lib/models/project.dart
class Project {
  final String title;
  final String description;
  final List<String> imageUrls;
  final String projectUrl;
  final List<String> tags;

  Project({
    required this.title,
    required this.description,
    required this.imageUrls,
    required this.projectUrl,
    required this.tags,
  });

  factory Project.fromJson(Map<String, dynamic> json) {
    return Project(
      title: json['title'] as String,
      description: json['description'] as String,
      imageUrls: List<String>.from(json['imageUrl'] as List<dynamic>),
      projectUrl: json['projectUrl'] as String,
      tags: List<String>.from(json['tags'] as List<dynamic>),
    );
  }
}
